marimo/_plugins/ui/_impl/table.py

```diff
@@ -322,20 +324,46 @@ def __init__(
         # (searching operations include query, sort, filter, etc.)
         self._searched_manager = self._manager
         # Holds the data after user selecting from the component
-        self._selected_manager: Optional[TableManager[Any]] = None
+        self._selected_manager: Optional[
+            Union[TableManager[Any], list[TableCell]]
+        ] = None
 
         self._selection = selection
         self._has_any_selection = False
-        initial_value = []
+        initial_value: Union[list[int], list[dict[str, Any]]] = []
         if initial_selection and self._manager.supports_selection():
-            if selection == "single" and len(initial_selection) > 1:
+            if (selection in ["single", "single-cell"]) and len(
+                initial_selection
+            ) > 1:
                 raise ValueError(
                     "For single selection mode, initial_selection can only contain one row index"
                 )
             try:
-                self._selected_manager = self._searched_manager.select_rows(
-                    initial_selection
-                )
+                if selection in ["single-cell", "multi-cell"]:
+                    self._selected_manager = (
+                        self._searched_manager.select_cells(
+                            [
+                                TableCoordinate(
+                                    row_id=v["rowId"],
+                                    column_name=v["columnName"],
+                                )
+                                for v in initial_selection
+                                if isinstance(v, dict)
+                                and "rowId" in v
```

Comment:
i think this might be better as a tuple instead of `rowId` and `columnId`. python developers use `snake_case` as well so it wouldn't be intuitive for them to construct dictionaries with `rowId/columnName`

Reply:
So something like
```py
        initial_selection: Optional[
            Union[list[int], list[tuple[str, str]]]
        ] = None,
```